File: flyweight/c-sharp/BlackBox.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Flyweight
{
    // Flyweight (Box) şablonunu uygulayan sınıftır.
    // UML diyagramındaki ConcreteFlyweight sınıfına denk gelmektedir.
    class BlackBox : Box
    {
        public BlackBox(int width, int height)
        {
            this.width = width;
            this.height = height;
            this.boxColor = BoxColor.Black;
        }

        public override void Draw(int locationX, int locationY)
        {
            Console.WriteLine($"{boxColor} box drawn. {locationX},{locationY}");
        }
    }
}

File: flyweight/typescript/black-box.ts
import {Box} from "./box";
import {BoxColor} from "./box-color";

// Flyweight (Box) şablonunu uygulayan sınıftır.
// UML diyagramındaki ConcreteFlyweight sınıfına denk gelmektedir.
export class BlackBox extends Box {
  constructor(readonly width, readonly height) {
    super();
    this.width = width;
    this.height = height;
    this.boxColor = BoxColor.BLACK;
  }

  public draw(locationX: number, locationY: number): void {
    console.log(`${this.boxColor} box drawn. ${locationX},${locationY}`)
  }
}
